src/NuGet.Protocol.Core.Types/SourcePackageDependencyInfo.cs

```diff
@@ -8,28 +9,100 @@ namespace NuGet.Protocol.Core.Types
     public class SourcePackageDependencyInfo : PackageDependencyInfo
     {
         public SourcePackageDependencyInfo(string id, NuGetVersion version, bool listed, SourceRepository source)
-            : base(id, version)
+            : this(
+                  id,
+                  version,
+                  dependencies: Enumerable.Empty<PackageDependency>(),
+                  listed: listed,
+                  source: source)
         {
-            Listed = listed;
-            Source = source;
         }
 
-        public SourcePackageDependencyInfo(PackageIdentity identity, IEnumerable<PackageDependency> dependencies, bool listed, SourceRepository source)
-            : base(identity, dependencies)
+        public SourcePackageDependencyInfo(
```

Comment:
Many of the parameters is optional, is there a reason not to have these constructors?